### PR TITLE
OEL-1014: Module scaffold

### PR DESCRIPTION
**Pull request by GilNovacomm**
Starter Template created from drupal-modules-template:
Updated Drone image to 8.1 (as per oe_contact_form PR).
Updated minimum Drupal version to 9.2.
Removed all references to Behat.
Updated code-review to 2.0
Updated docker-compose.
Updated README.md informing that we are not compatible with oe_content content types.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    when:
+      event: [ deployment ]
+
+workspace:
+  base: /test
+  path: oe_starter_content
+
+services:
+  web:
+    image: fpfis/httpd-php-ci:${PHP_VERSION}
```
Comment (drishu) [inline .drone.yml]: lets use the internal fpfis images like here https://github.com/openeuropa/oe_bootstrap_theme/blob/1.x/.drone.yml#L17, do this for all images

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    environment:
+      - DOCUMENT_ROOT=/test/oe_starter_content
+      - COMPOSERVER=--2
+  mysql:
+    image: percona/percona-server:5.6
```
Comment (drishu) [inline .drone.yml]: use this setup for mysql https://github.com/openeuropa/oe_bootstrap_theme/blob/1.x/.drone.yml#L21

File: .drone.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+pipeline:
+  composer-install-highest:
+    group: prepare
+    image: fpfis/httpd-php-ci:${PHP_VERSION}
+    volumes:
+      - /cache:/cache
+    commands:
+      - composer self-update --2
+      - composer install --ansi --no-progress
```
Comment (drishu) [inline .drone.yml:36]: lets leave the suggests out 
```suggestion
      - composer install --ansi --no-suggest --no-progress
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,124 @@
-# oe_starter_content
-OpenEuropa starter content types
+# OpenEuropa oe_starter_content
```
Comment (drishu) [inline README.md]:
```suggestion
# OpenEuropa Starter content
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,124 @@
...
+
+The OpenEuropa Starter Content provides a starting template for OpenEuropa content modules.
```
Comment (drishu) [inline README.md:3]: This bit you need to update to match what this repo is offering, use this for inspiration https://github.com/openeuropa/oe_content/blob/2.x/README.md, but one remark, these wont be "corporate" content types

File: composer.json [hunk 6]
```diff
@@ -0,0 +1,67 @@
+{
+    "name": "openeuropa/oe_starter_content",
+    "description": "OpenEuropa Drupal module template.",
```
Comment (drishu) [inline composer.json]:
```suggestion
    "description": "OpenEuropa Drupal module that contains the EC entity types.",
```

File: composer.json [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        }
+    ],
+    "autoload": {
+        "psr-4": {
+            "Drupal\\oe_starter_content\\": "./src/"
+        }
+    },
+    "autoload-dev": {
+        "psr-4": {
+            "Drupal\\Tests\\oe_starter_content\\": "./tests/"
```
Comment (drishu) [inline composer.json]: We need to force tests under src
```suggestion
            "Drupal\\Tests\\oe_starter_content\\": "./tests/src"
```

File: oe_starter_content.info.yml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+name: OpenEuropa Starter Content
+description: OpenEuropa Starter Content.
```
Comment (drishu) [inline oe_starter_content.info.yml]:
```suggestion
description: Contains the EC entity types in Drupal.
```

File: phpunit.xml.dist [hunk 9]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<phpunit bootstrap="${drupal.root}/core/tests/bootstrap.php" backupGlobals="true" colors="true" >
```
Comment (drishu) [inline phpunit.xml.dist]:
```suggestion
<phpunit bootstrap="${drupal.root}/core/tests/bootstrap.php" backupGlobals="true" colors="true" cacheResult="false">
```

File: phpunit.xml.dist [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+  <php>
+    <ini name="error_reporting" value="32767"/>
+    <ini name="memory_limit" value="-1"/>
+    <env name="SIMPLETEST_IGNORE_DIRECTORIES" value="${drupal.root}"/>
```
Comment (drishu) [inline phpunit.xml.dist]:
```suggestion
    <env name="SIMPLETEST_IGNORE_DIRECTORIES" value="vendor,node_modules,${drupal.root}"/>
```

File: phpunit.xml.dist [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+    <env name="SIMPLETEST_BASE_URL" value="${drupal.base_url}"/>
+    <env name="SIMPLETEST_DB" value="mysql://${drupal.database.user}:${drupal.database.password}@${drupal.database.host}:${drupal.database.port}/${drupal.database.name}"/>
+  </php>
+  <testsuites>
+    <testsuite>
```
Comment (drishu) [inline phpunit.xml.dist]: add a name

**Review by drishu — CHANGES_REQUESTED** [review]

File: .drone.yml (hunk 3 above)
Comment (brummbar) [reply .drone.yml:36]: Suggest is deprecated in composer 2 and will be removed from 3, so we don't use it anymore

File: README.md (hunk 5 above)
Comment (brummbar) [reply README.md:3]: I will do this.

File: .drone.yml [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+
+pipeline:
+  composer-install-highest:
+    group: prepare
+    image: registry.fpfis.eu/fpfis/httpd-php:${PHP_VERSION}-ci
+    volumes:
+      - /cache:/cache
+    commands:
+      - composer self-update --2
+      - composer install --ansi --no-suggest --no-progress
```
Comment (brummbar) [inline .drone.yml]: `--no-suggest` has been removed from oe_contact_forms because it's deprecated in composer 2.

File: .drone.yml [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+        COMPOSER_BOUNDARY: highest
+
+  composer-update-lowest:
+    group: prepare
+    image: registry.fpfis.eu/fpfis/httpd-php:${PHP_VERSION}-ci
+    volumes:
+      - /cache:/cache
+    commands:
+      - composer self-update --2
+      - composer update --prefer-lowest --prefer-stable --ansi --no-progress
```
Comment (brummbar) [inline .drone.yml:48]: Same as above.

**Review by brummbar — CHANGES_REQUESTED** [review]

File: .drone.yml (hunk 13 above)
Comment (GilNovacomm) [reply .drone.yml:48]: Should I remove `--no-progress`? There's no `--no-suggest` on this line.

Comment (drishu) [reply .drone.yml:48]: @GilNovacomm --no-progress stays, I found this document helpful https://getcomposer.org/upgrade/UPGRADE-2.0.md check it out

Comment (brummbar) [reply .drone.yml:48]: It was my bad, I read suggest instead of progress 😞

**Review by brummbar — APPROVED** [review]

**Review by drishu — APPROVED** [review]

**Pull request merged**